dre-service/src/main/java/io/cdap/re/RulesDB.java

```diff
@@ -386,9 +366,9 @@ void addRuleToRulebook(String rulebookId, String ruleId)
     }
 
     StructuredRow rulebookRow = optionalRulebookStructuredRow.get();
-    List<String> rules = convertRulesToSet(rulebookRow.getString(RULES_COL));
+    List<String> rules = GSON.fromJson(rulebookRow.getString(RULES_COL), LIST_STRING_TYPE);
 
-    if (rules.contains(ruleId)) {
+    if (rules == null || rules.contains(ruleId)) {
```

Comment:
is it possible to be null? If so, why would that mean the rule already exists?

Reply:
Yes, this is what I was not sure about since the line prior to this one is `List<String> rules = GSON.fromJson(rulebookRow.getString(RULES_COL), LIST_STRING_TYPE);`, which triggers a warning in IntelliJ that `rules` might be null. What is the best strategy to solve this possible null pointer? Not check and let the program crash? Check for null and if it is throw a RuntimeException?

Reply:
We need to distinguish between the fact that it can syntactically return null vs whether the logic allows it. If all the write methods ensure that something is written for rules, it can never be null.

Reply:
I agree that a null value for `rules` should be treated the same as an empty rules list. However, if `rules` is null and we call a method on that object, it will throw a NullPointerException. Should we worry about that exception? Or let the caller decide what to do if an exception occurs?